radixdlt-java/src/main/java/com/radixdlt/client/core/ledger/InMemoryAtomStore.java

```diff
@@ -108,9 +108,11 @@ void put(AtomObservation atomObservation) {
 						// That is, create soft deletes for any known missing deletes from dependent atoms
 						List<AtomObservation> missingDeletes = new ArrayList<>();
 						if (nextUpdate.getType() == Type.DELETE && include) {
+							// TODO: Should we DELETE atoms which are dependent on these atoms as well?
 							List<AtomObservation> observationsContainingDown = observation.getAtom().particles(Spin.UP)
 								.flatMap(p -> atomsObservationState.atomContainingDown(p)
-									.map(o -> o.isStore() ? Stream.of(o) : Stream.<AtomObservation>empty())
+									.map(o -> o.isStore() && !o.getAtom().getHid().equals(observation.getAtom().getHid())
```

Comment:
Side note: When the Core returns that an Atom has 'already' been stored, what should happen here? Should it be considered a 'store'? Or special case it?

Reply:
client just skips it, if its already been stored